common/src/main/java/com/wynntils/features/user/inventory/InventoryEmeraldCountFeature.java

```diff
@@ -0,0 +1,188 @@
+            emeraldText = String.valueOf(emeralds) + EmeraldSymbols.E;
+        } else {
+            String[] emeraldAmounts = Arrays.stream(calculateEmeraldAmounts(emeralds))
+                    .mapToObj(String::valueOf)
+                    .toArray(String[]::new);
+            emeraldText = emeraldAmounts[2] + EmeraldSymbols.LE + " " + emeraldAmounts[1] + EmeraldSymbols.EB + " "
+                    + emeraldAmounts[0] + EmeraldSymbols.E;
+        }
+
+        FontRenderer.getInstance()
+                .renderText(
+                        poseStack,
+                        emeraldText,
+                        containerScreen.leftPos + 1,
+                        containerScreen.topPos - 10,
+                        0,
+                        CommonColors.WHITE,
+                        HorizontalAlignment.Left,
+                        VerticalAlignment.Top,
+                        FontRenderer.TextShadow.NORMAL);
+
+        poseStack.popPose();
+    }
+
+    private void renderTexturedCount(PoseStack poseStack, int emeralds) {
+        Screen screen = McUtils.mc().screen;
+
+        if (!(screen instanceof AbstractContainerScreen<?> containerScreen)) return;
+
+        poseStack.pushPose();
+        poseStack.translate(containerScreen.leftPos, containerScreen.topPos, 0);
+
+        String[] emeraldAmounts = new String[3];
+        if (GLFW.glfwGetKey(McUtils.mc().getWindow().getWindow(), GLFW.GLFW_KEY_LEFT_SHIFT) == 1) {
+            emeraldAmounts[0] = StringUtils.formatAmount(emeralds);
+            emeraldAmounts[1] = StringUtils.formatAmount(emeralds / 64d);
+            emeraldAmounts[2] = StringUtils.formatAmount(emeralds / 4096d);
+        } else {
+            emeraldAmounts = Arrays.stream(calculateEmeraldAmounts(emeralds))
+                    .mapToObj(String::valueOf)
+                    .toArray(String[]::new);
+        }
+
+        int renderedCount = 0;
+
+        for (int i = emeraldAmounts.length - 1; i >= 0; i--) {
+            String emeraldAmount = emeraldAmounts[i];
+            if (Objects.equals(emeraldAmount, "0")) continue;
+
+            final int renderX = -TEXTURE_SIZE;
+            final int renderY = renderedCount * TEXTURE_SIZE;
+            RenderUtils.drawTexturedRect(
+                    poseStack,
+                    Texture.EMERALD_COUNT_BACKGROUND.resource(),
+                    renderX,
+                    renderY,
+                    0,
+                    TEXTURE_SIZE,
+                    TEXTURE_SIZE,
+                    0,
+                    0,
+                    Texture.EMERALD_COUNT_BACKGROUND.width(),
+                    Texture.EMERALD_COUNT_BACKGROUND.height(),
+                    Texture.EMERALD_COUNT_BACKGROUND.width(),
+                    Texture.EMERALD_COUNT_BACKGROUND.height());
+
+            Item item =
+                    switch (i) {
+                        case 2 -> Items.EXPERIENCE_BOTTLE;
```

Comment:
I think the associated Item type could be joined with the EmeraldSymbol enum. Maybe a more general EmeraldInfo class, or something like that. It could give you a symbol or Item type for every denomination (unit, block, liquid). It could also contain static helper methods such as  calculateEmeraldAmounts.